### ISSUE TITLE
some demo about Beam Search with LM

### ISSUE THREAD

**Issue by HuizhenShu** [ynop/py-ctc-decode#1]
thanks for this project.
I want to get the decoder result with"Beam Search with LM", can you give out some demo to show how to use it?

**Comment by ynop** [comment]
I extended the example in the Readme a bit.
https://github.com/ynop/py-ctc-decode#beam-search-with-lm

Hope it helps.
Otherwise just ask.

**Comment by HuizhenShu** [comment]
Thanks for your reply!
my test logits' shape  is (1, 131, 1224)(batch_size,time_step,V), the true result is a sentence with 20 words. But what I got from decode_batch has only one word.

**Comment by ynop** [comment]
* Are you sure that the given sentence is predicted with the language model you use? Words not in the language model won't be predicted.
* Have you used ' ' as space and '_' as blank in the vocabulary?
* Have you used log probabilities?

Otherwise maybe try best-path to check if it works there.

**Comment by HuizhenShu** [comment]
> * Are you sure that the given sentence is predicted with the language model you use? Words not in the language model won't be predicted.
> * Have you used ' ' as space and '_' as blank in the vocabulary?
> * Have you used log probabilities?
> 
> Otherwise maybe try best-path to check if it works there.

**Comment by HuizhenShu** [comment]
Thanks for your reply!
1. yes, I use the same corpus to train the language model and the acoustic model

2. '_' is in my vocabulary, but ' ' is not

3.  log probabilities? Are you mean when compute the ctc-loss,  transpose the logits with tf.log first? if so ,yes,I used it.

I have try the  best-path method. what I got is a bunch of words without blank in them. Then, I add the blank in BestPathDecoder.decode[line15] : pred = ' '.join(pred).replace('_', '') ,I got the reasonable result.
Should I retrain my acoustic model with a new vocabulary( the version which adds ' ')

**Comment by HuizhenShu** [comment]
The corpus I use is like this -->'zhe4 feng1 xin4 xie3 yu2 gong1 yuan2 yi1 liu4 wu3 si4 nian2 shi4 '
I would split the sentence and transform the words into ids, so, theoretically，space is not a part of the input data. In this case, can I use your【Beam Search with LM】 in some way?

**Comment by ynop** [comment]
Hmm, the space is needed for since it is the point which triggers the language model.
If your input are word ids you would have to adapt the algorithm.

**Comment by HuizhenShu** [comment]
I seem to understand. I will try these two methods below
 1. retrain the acoustic model with new data (add a space id in the middle of each word id )
 2. when use the 【Beam Search with LM】,add space to each symbol I got.
Thank you for your patience. I'll reply this when I get results

**Comment by HuizhenShu** [comment]
I have try the second method. Add a space before symbol when calculate value. It works.
Thanks a lot
